scripts/build.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+# Change to the root of the repo
+cd "${BASH_SOURCE%/*}/.."
+
+# dist
+tsc
+rsync -r --exclude='*.ts' --exclude='*.js' src/ dist/
+
+# esnext
+tsc -p tsconfig.esnext.json
+rsync -r --exclude='*.ts' --exclude='*.js' src/ esnext/
```

Comment:
Doesn't running `tsc` produce `.js.map`, `.d.ts`, and `.js` files? Why are we excluding `.ts` and `.js` files?

Reply:
Rsync is being used to transfer any non `.js` or .`ts` files to the `dist` and `esnext` folders. tsc handles the files excluded by rsync. It's technically not needed right now, but it's future proofing incase we do need it.